peek: fails to build on macOS

**Describe the bug**
peek fails to build on macOS

**To Reproduce**
Clone `nixpkgs` then `nix-build -A peek`

**Notify maintainers**
@puffnfresh @worldofpeace 

**Log**
```
keybinder-3.0 v0.3.2

Prefix............... : /nix/store/8zd2sdjmbpw9c9rsk88g1nxjvjbhw0w9-keybinder3-0.3.2
GTK version ......... : gtk+-3.0
libkeybinder ........ : yes
gobject-introspection : yes
gtk-doc ............. : no

Now type make to compile
Then su to root and type: make install

building
build flags: SHELL=/nix/store/saggdc3mgg1h6fs05r9rv5isypqwzby8-bash-4.4-p23/bin/bash
Making all in libkeybinder
make[1]: Entering directory '/private/var/folders/s1/sm71lqk553d1_f8drd3tz_5r0000gn/T/nix-build-keybinder3-0.3.2.drv-0/source/libkeybinder'
  CC     bind.lo
bind.c:30:10: fatal error: 'gdk/gdkx.h' file not found
#include <gdk/gdkx.h>
         ^~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:424: bind.lo] Error 1
make[1]: Leaving directory '/private/var/folders/s1/sm71lqk553d1_f8drd3tz_5r0000gn/T/nix-build-keybinder3-0.3.2.drv-0/source/libkeybinder'
make: *** [Makefile:359: all-recursive] Error 1
builder for '/nix/store/kpd2gm2yfk840b18yi9yfxhd66klpbc2-keybinder3-0.3.2.drv' failed with exit code 2
cannot build derivation '/nix/store/vz8jh6pj5h6sd9w2s19s85ignk10y5ba-peek-1.5.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/vz8jh6pj5h6sd9w2s19s85ignk10y5ba-peek-1.5.1.drv' failed
```

--- a/pkgs/development/libraries/gtk/3.x.nix
+++ b/pkgs/development/libraries/gtk/3.x.nix
@@ -148,6 +148,8 @@ stdenv.mkDerivation rec {
     "-Dtests=false"
     "-Dtracker3=${lib.boolToString trackerSupport}"
     "-Dbroadway_backend=${lib.boolToString broadwaySupport}"
+    "-Dx11_backend=${lib.boolToString x11Support}"
+    "-Dquartz_backend=${lib.boolToString (stdenv.isDarwin && !x11Support)}"
   ];
 
   doCheck = false; # needs X11
@@ -159,6 +161,10 @@ stdenv.mkDerivation rec {
   NIX_CFLAGS_COMPILE = "-DG_ENABLE_DEBUG -DG_DISABLE_CAST_CHECKS";
 
   postPatch = ''
+    # See https://github.com/NixOS/nixpkgs/issues/132259
+    substituteInPlace meson.build \
+      --replace "x11_enabled = false" ""
+
     files=(
       build-aux/meson/post-install.py
       demos/gtk-demo/geninclude.py

--- a/pkgs/development/libraries/keybinder3/default.nix
+++ b/pkgs/development/libraries/keybinder3/default.nix
@@ -27,7 +27,7 @@ stdenv.mkDerivation rec {
     description = "Library for registering global key bindings";
     homepage = "https://github.com/kupferlauncher/keybinder/";
     license = licenses.mit;
-    platforms = platforms.linux;
+    platforms = platforms.unix;
     maintainers = [ maintainers.cstrahan ];
   };
 }

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17002,6 +17002,7 @@ with pkgs;
   };
 
   keybinder3 = callPackage ../development/libraries/keybinder3 {
+    gtk3 = if stdenv.isDarwin then gtk3-x11 else gtk3;
     automake = automake111x;
   };
 

